lib/checkother.cpp

```diff
@@ -3308,43 +3349,32 @@ void CheckOther::checkEvaluationOrder()
                 if (parent->str() == "(" && parent->astOperand2())
                     break;
 
-                // self assignment..
-                if (tok2 == tok &&
-                    tok->str() == "=" &&
-                    parent->str() == "=" &&
-                    isSameExpression(false, tok->astOperand1(), parent->astOperand1(), *mSettings, true, false)) {
-                    if (mSettings->severity.isEnabled(Severity::warning) &&
-                        isSameExpression(true, tok->astOperand1(), parent->astOperand1(), *mSettings, true, false))
-                        selfAssignmentError(parent, tok->astOperand1()->expressionString());
-                    break;
-                }
-
-                // Is expression used?
-                bool foundError = false;
-                visitAstNodes((parent->astOperand1() != tok2) ? parent->astOperand1() : parent->astOperand2(),
-                              [&](const Token *tok3) {
-                    if (tok3->str() == "&" && !tok3->astOperand2())
-                        return ChildrenToVisit::none; // don't handle address-of for now
-                    if (tok3->str() == "(" && Token::simpleMatch(tok3->previous(), "sizeof"))
-                        return ChildrenToVisit::none; // don't care about sizeof usage
-                    if (isSameExpression(false, tok->astOperand1(), tok3, *mSettings, true, false))
-                        foundError = true;
-                    return foundError ? ChildrenToVisit::done : ChildrenToVisit::op1_and_op2;
-                });
+                bool foundError{false}, foundUnspecified{false}, bSelfAssignmentError{false};
+                if (mTokenizer->isCPP() && mSettings->standards.cpp >= Standards::CPP17)
+                    foundError = checkEvaluationOrderPost17(tok, tok2, parent, *mSettings, foundUnspecified);
```

Comment:
"Post17" sounds unfortunate because it is executed for 17. How about the name "checkEvaluationOrderCpp17".

I don't think "Pre11" is a good name neither. This should be executed for C code. It is the old C rules that are used so why not "checkEvaluationOrderC".

The "checkEvaluationOrderC" should not be executed for C++11 and C++14 there are important differences.